app/src/main/java/ca/marshallasch/veil/database/Migrations.java

```diff
@@ -15,7 +15,15 @@
      * In this version change the users table was added.
      * @param db the underlying database object
      */
-    static void v1ToV3(SQLiteDatabase db) {
+    /*static void upgradeV3(SQLiteDatabase db) {
         db.execSQL(UserContract.SQL_CREATE_USERS);
+    }*/
+
+    /**
+     * In this version change the known posts table was added and the known hashes table was dropped
+     * @param db the underlying database object
+     */
+    static void upgradeV4(SQLiteDatabase db){
+        db.execSQL(KnownPostsContract.SQL_CREATE_KNOWN_HASHES);
```

Comment:
you removed the table but you did not remove it in the migration.

Reply:
removed KnownHashes table in migration.